src/storage/src/hummock/local_version.rs

```diff
@@ -49,38 +48,39 @@ pub struct LocalVersion {
 pub enum SyncUncommittedData {
     /// Before we start syncing, we need to mv data from shared buffer to `sync_uncommitted_data`
     /// as `Syncing`.
-    Syncing(Vec<KeyIndexedUncommittedData>),
+    Syncing(OrderSortedUncommittedData),
     /// After we finish syncing, we changed `Syncing` to `Synced`.
     Synced(Vec<LocalSstableInfo>),
 }
 
 impl SyncUncommittedData {
-    pub fn get_overlap_data<R, B>(&self, key_range: &R) -> OrderSortedUncommittedData
+    pub fn get_overlap_data<R, B>(
+        &self,
+        key_range: &R,
+        epoch: HummockEpoch,
+    ) -> OrderSortedUncommittedData
     where
         R: RangeBounds<B>,
         B: AsRef<[u8]>,
     {
         match self {
             SyncUncommittedData::Syncing(task) => {
-                let local_data_iter = task
+                vec![task
```

Comment:
It is better to keep the nested vec structure instead of flattening it so that we can only use ordered merge iterator for the inner vec.

Reply:
We'd better use 
```
task.iter().map(|data_vec| {
    data_vec.iter().filter(|data| {
        // filter logic
    })
    .collect_vec()
})
```

so that the order of original data structure is unchanged.

Reply:
okokokk